test/functional/Valhalla/src/org/openj9/test/lworld/ValueTypeUnsafeTests.java

```diff
@@ -29,39 +29,67 @@
 import java.util.List;
 import static org.testng.Assert.*;
 import org.testng.annotations.Test;
+import org.testng.annotations.BeforeMethod;
 import org.testng.annotations.BeforeClass;
 import static org.openj9.test.lworld.ValueTypeUnsafeTestClasses.*;
 
 
 @Test(groups = { "level.sanity" })
 public class ValueTypeUnsafeTests {
-	static Unsafe myUnsafe = null;
+	static Unsafe myUnsafe = Unsafe.getUnsafe();
 	static boolean isCompressedRefsEnabled = false;
 	static boolean isFlatteningEnabled = false;
 	static boolean isArrayFlatteningEnabled = false;
 
-	@BeforeClass
-	static public void testSetUp() throws RuntimeException {
-		myUnsafe = Unsafe.getUnsafe();
+	static ValueTypePoint2D vtPoint;
+	static ValueTypePoint2D[] vtPointAry;
+	static long vtPointOffsetX;
+	static long vtPointOffsetY;
+	static long vtLongPointOffsetX;
+	static long vtLongPointOffsetY;
+	static long vtPointAryOffset0;
+	static long vtPointAryOffset1;
+	static long intWrapperOffsetI;
 
+	@BeforeClass
+	static public void testSetUp() throws Throwable {
 		List<String> arguments = ManagementFactory.getRuntimeMXBean().getInputArguments();
 		isFlatteningEnabled = arguments.contains("-XX:ValueTypeFlatteningThreshold=99999");
 		isArrayFlatteningEnabled = arguments.contains("-XX:+EnableArrayFlattening");
 		isCompressedRefsEnabled = arguments.contains("-Xcompressedrefs");
+
+		vtPointOffsetX = myUnsafe.objectFieldOffset(ValueTypePoint2D.class.getDeclaredField("x"));
+		vtPointOffsetY = myUnsafe.objectFieldOffset(ValueTypePoint2D.class.getDeclaredField("y"));
+		vtLongPointOffsetX = myUnsafe.objectFieldOffset(ValueTypeLongPoint2D.class.getDeclaredField("x"));
+		vtLongPointOffsetY = myUnsafe.objectFieldOffset(ValueTypeLongPoint2D.class.getDeclaredField("y"));
+		vtPointAryOffset0 = myUnsafe.arrayBaseOffset(ValueTypePoint2D[].class);
+		intWrapperOffsetI = myUnsafe.objectFieldOffset(IntWrapper.class.getDeclaredField("i"));
+	}
+
+	static private <T> long arrayElementSize(T[] array) {
+		// don't simply use getObjectSize(ary[0]) because the amount of memory used to store an object may be different depending on if it is inside an array or not
```

Comment:
What I meant is that using `myUsafe.getObjectSize(array[0]) - myUnsafe.valueHeaderSize(array[0].getClass())` will not give the correct array element size in some cases, such as when flattening is disabled.
I can change it to _"don't simply use getObjectSize(array[0]) - valueHeaderSize(array[0].getClass()) because the amount of memory used to store an object may be different depending on if it is inside an array or not"_ if you think that would be more clear?

Reply:
> such as when flattening is disabled.

I suggest to also include it.

Reply:
added in c83a184